routers/window_usage/query_creation.py

```diff
@@ -0,0 +1,150 @@
+from beanie import init_beanie
+from beanie.operators import ElemMatch, LTE, GTE
+import motor
+from typing import Dict, List
+
+from routers.window_usage.models import (AggregationFieldEnum,
+                                         WindowActivationEvents,
+                                         WindowUsageData,
+                                         WindowUsageQueryParameters)
+
+
+async def fetch_window_activation_events(
+        query_params: WindowUsageQueryParameters):
```

Comment:
It is fine behavior to pass arguments in one object to a function, though, I don't think the function signature should be effected. Function signature should have what the function requires (hence, required arguments, required key arguments).
```suggestion
        company_id: UUID4,
        team_id: UUID4,
        min_date: datetime,
        max_date: datetime,
        tag_filters: List[Tag]
```

Unless the parameter is actually an object, then passing the object is fine. `WindowUsageQueryParameters` is more of a container. One thing could work and has a nice compromise, continuing on the code above:

```
await fetch_window_activation_events(**WindowUsageQueryParametersInstance.dict())
```

Then you can generate the query parameter as object, but the function signature is descriptive. I look at the signature and it hit me in the face what this function is doing.

Reply:
This change is not only for aesthetic reasons, even though, python is a language with emphasis on readability. If the query object is constructed missing one argument, say "company_id" due to a bug. This function will still trigger happily, until it hit `something = query_params.company_id`, which will raise `AttributeError`, at that point, good luck finding if `query` object has mutated or if it had the `company_id` to start with.

```
def bar(foo):
    return foo

bar()  # Raises MissingArgument
```